crt/fork.c

```diff
@@ -122,13 +122,13 @@ struct pthread* _create_child_pthread_and_copy_stack(
      * parent process exit relies specific logic during child process exit to
      * clear the ownership indication. If any part of the relevant design
      * changes, the implementaiton needs to be reconsidered */
-    if (!(map = mmap(
-              NULL,
-              size_rounded,
-              PROT_READ | PROT_WRITE,
-              MAP_ANONYMOUS,
-              -1,
-              0)))
+    if ((map = mmap(
+             NULL,
+             size_rounded,
+             PROT_READ | PROT_WRITE,
+             MAP_ANONYMOUS,
```

Comment:
Suggest adding ``MAP_PRIVATE`` too.

Reply:
ok

Reply:
@paulcallen you've added MAP_PRIVATE to the `prot`, instead of `flags`!